### PR TITLE
CircularBuffer

### PR DESCRIPTION
**Pull request by kmsquire**
Small patch to make tests in #144 pass.  Closes #144.

Cc: @tbreloff

### CODE REVIEW

File: src/circularbuffer.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    first::Int
+    buffer::Vector{T}
+
+    CircularBuffer(capacity::Int) = new(capacity, 1, T[])
+end
+
+function _buffer_index(cb::CircularBuffer, i::Int)
+    n = length(cb)
+    if i < 1 || i > n
+        error("CircularBuffer out of range. cb=$cb i=$i")
```
Comment (simonster) [inline src/circularbuffer.jl]: This should still be a BoundsError.

Comment (kmsquire) [inline src/circularbuffer.jl]: Fixed, thanks!

**Comment by kmsquire** [comment]
This has been around for a while, first as #109, then #144, so there's been quite a bit of time for commenting already.

Since I haven't commented myself before... `DataStructures` has kind of become a catch-all for various general data structures that people need/want, but the package itself doesn't have a roadmap or philosophy about what belongs here and what form the things that do go here should have.  

I'd like that to change, but with my uncertainty of my own availability, there's no guarantee that I will make this will happen in a reasonable time frame.  

So if anyone has a burning desire to gain fame and glory by whipping `DataStructures` into shape, please speak up... ;-)

**Comment by kmsquire** [comment]
Oh, and I'll merge tomorrow if there are no further comments.

**Comment by kmsquire** [comment]
Thanks Daniel!

**Pull request merged**